dot/core/service.go

```diff
@@ -102,11 +96,7 @@ func (s *Service) Start() error {
 
 // Stop stops the core service
 func (s *Service) Stop() error {
-	s.Lock()
-	defer s.Unlock()
-
-	s.cancel()
-	close(s.blockAddCh)
+	close(s.stopCh)
```

Comment:
Kinda out of scope, but I couldn't hold myself, done in 592d4cbd95e9f0a5b1cb4bb8a455a1bf0b89ba73

Note I changed `handleBlocksAsync` to be `handleBlocksAsync(stop <-chan struct{}, done chan<- struct{})` to force only consumption/production on each channel, since I have seen horrible things in the past I really want to avoid!